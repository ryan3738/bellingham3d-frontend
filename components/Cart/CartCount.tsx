import { CSSTransition, TransitionGroup } from 'react-transition-group';
import styled from 'styled-components';

const Dot = styled.div`
  color: black;
  padding: 0.5rem;
  /* Makes all numbers the same width. Two different browser supports */
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;
`;

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: transform 0.4s;
    backface-visibility: hidden;
  }
  .count-enter {
    transform: scale(4) rotateX(0.5turn);
  }

  .count-enter-active {
    transform: rotateX(0);
  }
  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0);
  }
  .count-exit-active {
    transform: scale(4) rotateX(0.5turn);
  }
`;

export default function CartCount({ count }: { count: number }): JSX.Element {
  return (
    <AnimationStyles>
      <TransitionGroup>
        <CSSTransition
          unmountOnExit
          className="count"
          classNames="count"
          key={count}
          timeout={{ enter: 400, exit: 400 }}
        >
          <Dot>{count}</Dot>
        </CSSTransition>
      </TransitionGroup>
    </AnimationStyles>
  );
}
