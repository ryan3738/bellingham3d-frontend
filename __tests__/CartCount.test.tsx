import { render } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/Cart/CartCount';

describe('<CartCount />', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Matches snapshop', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it('Updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={11} />);
    expect(container.textContent).toEqual('11');
    // expect(container).toHaveTextContent('11');
    // Update the props
    rerender(<CartCount count={12} />);
    // Wait for __ ms
    expect(container.textContent).toEqual('1211');
    await wait(400);
    expect(container.textContent).toEqual('12');
    expect(container).toMatchSnapshot();
    debug();
  });
});
