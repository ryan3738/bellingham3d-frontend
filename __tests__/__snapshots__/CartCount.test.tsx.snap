// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches snapshop 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1n06z5c-1 gxwNxv"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1n06z5c-0 eXpjqW count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount /> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1n06z5c-1 gxwNxv"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1n06z5c-0 eXpjqW count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
